client/internal/httpsender_test.go

```diff
@@ -356,3 +357,57 @@ func TestPackageUpdatesWithError(t *testing.T) {
 
 	cancel()
 }
+
+func TestHTTPSenderSetProxy(t *testing.T) {
```

Comment:
Is there any way we can also add tests that use an actual proxy and connect through it? I don't know if there is a premade proxy library in Go we can use for that.

Reply:
OK, would be great to add it for http transport for now and skip for websockets.

Reply:
added below